frontend/pages/events.tsx

```diff
@@ -39,6 +52,161 @@ const CardList = styled.div`
   }
 `
 
+const localizer = momentLocalizer(moment)
+
+const CalendarEvent = ({
+  event: { resource },
+}: {
+  event: { resource: ClubEvent }
+}) => {
+  return (
+    <>
+      {resource.name} - {resource.club_name}
+    </>
+  )
+}
+
+enum EventsViewOption {
+  LIST = 'LIST',
+  CALENDAR = 'CALENDAR',
+}
+
+enum CalendarNavigation {
+  PREVIOUS = 'PREV',
+  NEXT = 'NEXT',
+  TODAY = 'TODAY',
+  DATE = 'DATE',
+}
+
+enum CalendarView {
+  MONTH = 'month',
+  WEEK = 'week',
+  WORK_WEEK = 'work_week',
+  DAY = 'day',
+  AGENDA = 'agenda',
+}
+interface CalendarHeaderProps {
+  date: Date
+  label: string
+  onNavigate(action: CalendarNavigation, newDate?: Date): void
+  onView(view: CalendarView): void
+  view: string
+  views: CalendarView[]
+}
+
+const StyledHeader = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin: 20px 0;
+  .tools {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    margin: 0;
+    .view-label {
+      font-size: 18px;
+    }
+    & > *:not(:first-child) {
+      margin-left: 20px;
+    }
+  }
+`
+
+const ViewContext = createContext<
+  [
+    option: EventsViewOption,
+    setOption?: Dispatch<SetStateAction<EventsViewOption>>,
+  ]
+>([EventsViewOption.CALENDAR])
+
+const EventsViewSwitcher = ({ viewOption, setViewOption }) => (
+  <div className="buttons has-addons mt-0 mb-0">
+    <button
+      id="event-view-list"
+      className={`button is-medium ${
+        viewOption === EventsViewOption.LIST ? 'is-selected is-info' : ''
+      }`}
+      onClick={() => {
+        setViewOption(EventsViewOption.LIST)
+      }}
+    >
+      <Icon name="grid" />
+    </button>
+    <button
+      id="event-view-calendar"
+      className={`button is-medium ${
+        viewOption === EventsViewOption.CALENDAR ? 'is-selected is-info' : ''
+      }`}
+      onClick={() => {
+        setViewOption(EventsViewOption.CALENDAR)
+      }}
+    >
+      <Icon name="calendar" />
+    </button>
+  </div>
+)
+
+const CalendarHeader = ({
+  label,
+  onNavigate,
+  onView,
+  view,
+}: CalendarHeaderProps) => {
+  const [viewOption, setViewOption] = useContext(ViewContext)
+  const _views: CalendarView[] = [
+    CalendarView.DAY,
+    CalendarView.WEEK,
+    CalendarView.MONTH,
+  ]
+  return (
+    <StyledHeader>
+      <Title className="title" style={{ color: CLUBS_GREY, margin: 0 }}>
+        Events
+      </Title>
+      <div className="tools">
+        <div className="view-label">{label}</div>
+        <div className="buttons has-addons mt-0 mb-0">
+          <button
+            className={`button is-medium`}
```

Comment:
@esinx You can just do `className="button is-medium"` in these cases.

Minor issue, can merge without this change.